src/main/java/me/champeau/gradle/japicmp/archive/Version.java

```diff
@@ -0,0 +1,145 @@
+package me.champeau.gradle.japicmp.archive;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+public class Version implements Comparable<Version>, Serializable {
+  private final SemanticVersion version;
+
+  public Version(String version) {
+    SemanticVersion v;
+    try {
+      v = SemanticVersion.semanticVersion(version);
+    } catch (VersionParseException e) {
+      v = new SemanticVersion(1, 0, null, null);
+    }
+    this.version = v;
+  }
+
+  @Override
+  public int compareTo(Version version) {
+    return this.version.compareTo(version.version);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    Version version1 = (Version) o;
+    return Objects.equals(version, version1.version);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(version);
+  }
+
+  @Override
+  public String toString() {
+    return version.toString();
+  }
+
+  public static boolean checkIsVersion(String rawVersion) {
+    try {
+      SemanticVersion.semanticVersion(rawVersion);
+    } catch (VersionParseException e) {
+      return false;
+    }
+    return true;
+  }
+
+  private static class SemanticVersion implements Comparable<SemanticVersion>, Serializable {
+    private final Integer era;
+    private final Integer major;
+    private final Integer minor;
+    private final Integer patch;
+
+    public SemanticVersion(Integer era, Integer major, Integer minor, Integer patch) {
+      this.era = era;
+      this.major = major;
+      this.minor = minor;
+      this.patch = patch;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+      SemanticVersion that = (SemanticVersion) o;
+      return Objects.equals(era, that.era)
+          && Objects.equals(major, that.major)
+          && Objects.equals(minor, that.minor)
+          && Objects.equals(patch, that.patch);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(era, major, minor, patch);
+    }
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder();
+      if (era != null) {
+        sb.append(era);
+      }
+      if (major != null) {
+        sb.append(".").append(major);
+      }
+      if (minor != null) {
+        sb.append(".").append(minor);
+      }
+      if (patch != null) {
+        sb.append(".").append(patch);
+      }
+      return sb.toString();
```

Comment:
What is expected when era=null, major=1, minor=1, patch=null?

Reply:
Done